src/common/multi_file_reader.cpp

```diff
@@ -387,4 +452,108 @@ void UnionByName::CombineUnionTypes(const vector<string> &col_names, const vecto
 	}
 }
 
+bool MultiFileReaderOptions::AutoDetectHivePartitioningInternal(const vector<string> &files) {
+	std::unordered_set<string> uset;
+	idx_t splits_size;
+	{
```

Comment:
nitpick: whats the purpose of this code block here? I think code blocks (curly braces without an if/else/etc in front) should only be used when they serve a goal: i.e. control the scope of variables. I don't think thats the case here and would therefore probably not use one!